VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,130 @@
+ */
+public class DiscussionTextUtilsTest extends BaseTestCase {
+
+    private TextView dummyTextView;
+    private Runnable dummyClickListener;
+
+    @Before
+    public void initValues() {
+        dummyTextView = new TextView(context);
+        dummyClickListener = new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Author clicked");
+            }
+        };
+    }
+
+    @Test
+    public void test_DuplicateWhitespacesRemoval() {
+        String input = "  start  end  ";
+        String expected = " start end ";
+        String output = DiscussionTextUtils.removeDuplicateWhitespaces(input).toString();
+        assertEquals(expected, output);
+    }
+
+    @Test
+    public void test_TextInTextViewNotNull() {
+        DiscussionTextUtils.setAuthorAttributionText(dummyTextView,
+                DiscussionTextUtils.AuthorAttributionLabel.POST,
+                new IAuthorDataImpl("testAuthor", "testLabel", new Date()),
+                dummyClickListener);
+        assertNotNull(dummyTextView.getText());
+    }
+
+    @Test
+    public void test_AuthorIsTappable() {
+        DiscussionTextUtils.setAuthorAttributionText(dummyTextView,
+                DiscussionTextUtils.AuthorAttributionLabel.POST,
+                new IAuthorDataImpl("testAuthor", "testLabel", new Date()),
+                dummyClickListener);
+        SpannableString cs = (SpannableString) dummyTextView.getText();
+        ClickableSpan[] spans = cs.getSpans(0, cs.length(), ClickableSpan.class);
+        if (config.isUserProfilesEnabled()) {
+            assertTrue(spans.length > 0);
+        } else {
+            assertTrue(spans.length == 0);
+        }
+    }
+
+    @Test
+    public void test_AuthorIsTappable_WhenNoCreationDate() {
+        DiscussionTextUtils.setAuthorAttributionText(dummyTextView,
+                DiscussionTextUtils.AuthorAttributionLabel.POST,
+                new IAuthorDataImpl("testAuthor", "testLabel", null),
+                dummyClickListener);
+        ClickableSpan[] spans = getClickableSpans();
+        if (config.isUserProfilesEnabled()) {
+            assertTrue(spans.length > 0);
+        } else {
+            assertTrue(spans.length == 0);
+        }
+    }
+
+    @Test
+    public void test_AuthorIsNotTappable_WhenNoAuthor() {
+        DiscussionTextUtils.setAuthorAttributionText(dummyTextView,
+                DiscussionTextUtils.AuthorAttributionLabel.POST,
+                new IAuthorDataImpl(null, "testLabel", new Date()),
+                dummyClickListener);
+        ClickableSpan[] spans = getClickableSpans();
+        assertTrue(spans.length == 0);
+    }
+
+    private ClickableSpan[] getClickableSpans() {
+        SpannableString spannableString = (SpannableString) dummyTextView.getText();
+        return spannableString.getSpans(0, spannableString.length(), ClickableSpan.class);
+    }
+
+    private static class IAuthorDataImpl implements IAuthorData {
```

Comment:
`StubAuthorData` sounds better.

>(As an aside, we don't need to prefix interfaces with I that's not really a java thing).

It isn't but it does make it easy to conclude that a class is an interface just by looking at the name.